Gifski/Tooltip.swift

```diff
@@ -0,0 +1,93 @@
+import AppKit
+
+final class Tooltip: NSPopover {
+	init(text: String, closeOnClick: Bool = true, contentInsets: NSEdgeInsets = .zero, maxWidth: CGFloat? = nil) {
+		super.init()
+		setupContent(text: text, closeOnClick: closeOnClick, contentInsets: contentInsets, maxWidth: maxWidth)
+	}
+
+	required init?(coder: NSCoder) {
+		super.init(coder: coder)
+		setupContent(text: "", closeOnClick: true, contentInsets: .zero, maxWidth: nil)
+	}
+
+	private func setupContent(text: String, closeOnClick: Bool, contentInsets: NSEdgeInsets, maxWidth: CGFloat?) {
+		contentViewController = ToolTipViewController(text: text, contentInsets: contentInsets, maxWidth: maxWidth) { [weak self] in
+			if closeOnClick {
+				self?.close()
+			}
+		}
+		behavior = closeOnClick ? .transient : .applicationDefined
+	}
+}
+
+fileprivate final class ToolTipViewController: NSViewController {
+	fileprivate var text: String
+	fileprivate var contentInsets: NSEdgeInsets
+	fileprivate var maxWidth: CGFloat?
+	fileprivate var onClick: (() -> Void)?
+
+	private var clickRecognizer: NSClickGestureRecognizer?
+
+	init(text: String, contentInsets: NSEdgeInsets, maxWidth: CGFloat?, onClick: (() -> Void)?) {
+		self.text = text
+		self.contentInsets = contentInsets
+		self.maxWidth = maxWidth
+		self.onClick = onClick
+		super.init(nibName: nil, bundle: nil)
+	}
+
+	required init?(coder: NSCoder) {
+		self.text = ""
+		self.contentInsets = .zero
+		super.init(coder: coder)
+	}
+
+	override func loadView() {
+		let wrapperView = NSView()
```

Comment:
Why is a wrapper view needed?

Reply:
I use it for insets - do you if I can do it without the wrapper view?

Reply:
You can set auto-layout constraints on `textField` against the superview, with inset. I thought that's what the `pinToSuperview` thing was for?

Reply:
True, I tried that, but when pinned to the superview wrapper it wasn't looking good - probably it has it's own position/layout (e.g. margin-left to pad it from the arrow).

Reply:
Ok. Doesn't matter then. A wrapper view is fine. Just wanted to make sure it was actually needed.